docs/source/extension/extension_multiple_ui.rst

```diff
@@ -0,0 +1,277 @@
+      autoStart: true,
+      // The IStatusBar is marked optional here. If it's available, it will
+      // be provided to the plugin as an argument to the activate function
+      // (shown below), and if not it will be null.
+      optional: [IStatusBar],
+      // Make sure to list any 'requires' and 'optional' features as arguments
+      // to your activate function (activate is always passed an Application,
+      // then required arguments, then optional arguments)
+      activate: (app: JupyterFrontEnd, statusBar: IStatusBar | null) => {
+        // ... Extension code ...
+      }
+    };
+
+This plugin marks ``IStatusBar`` as optional, and adds an argument for it to the
+plugin's ``activate`` function (which will be called by JupyterLab when the extension
+loads). If ``IStatusBar`` is not available, the second argument to the ``activate``
+function will be ``null``, as is the case when the extension is loaded in Jupyter
+Notebook 7.
+
+The extension always creates a common main widget, but when it comes time to use the
+status bar, the extension first checks if the ``IStatusBar`` is available, and only then
+proceeds to create a status bar item. This allows the extension to run successfully in both
+JupyterLab and Jupyter Notebook 7:
+
+.. code:: TypeScript
+
+    // Create a ShoutWidget and add it to the interface in the right sidebar
+    const shoutWidget: ShoutWidget = new ShoutWidget();
+    shoutWidget.id = 'JupyterShoutWidget'; // Widgets need an id
+
+    app.shell.add(shoutWidget, 'right');
+
+    // Check if the status bar is available, and if so, make
+    // a status bar widget to hold some information
+    if (statusBar) {
+      const statusBarWidget = new ShoutStatusBarSummary();
+
+      statusBar.registerStatusItem('shoutStatusBarSummary', {
+        item: statusBarWidget
+      });
+
+      // Connect to the messageShouted to be notified when a new message
+      // is published and react to it by updating the status bar widget.
+      shoutWidget.messageShouted.connect((widget: ShoutWidget, time: Date) => {
+        statusBarWidget.setSummary(
+          'Last Shout: ' + widget.lastShoutTime?.toString() ?? '(None)'
+        );
+      });
+    }
+
+Using Required Features to Switch Behaviors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another pattern you can follow is to export a list of plugins from your
+extension, then use different "requires" features to select different
+behaviors based on which app the extension is currently running in.
+
+Here is a snippet from `this sample extension <https://github.com/jupyterlab/extension-examples/tree/main/clap-button-message>`_
+which adds a *clap* button to the top area in JupyterLab, or to the
+right sidebar in Jupyter Notebook 7 (you can read the full extension
+example code there):
+
+.. code:: TypeScript
+
+    /**
+     * Initialization data for the @jupyterlab-examples/clap-button JupyterLab extension.
```

Comment:
This code snippets do not read great on more narrow screens, with some strings and comments requiring scrolling. Here is one case that might be easy cut:

```suggestion
     * Data for the @jupyterlab-examples/clap-button JupyterLab plugin.
```

Note `plugin` is more correct here than `extension` in technical sense.